src/UnkObjects/unk8600F8.cpp

```diff
@@ -25,3 +26,201 @@ void unk_8600F8::loc_611DC0() {
     *x = this->unk14;
     this->unk14 = nullptr;
 }
+
+unk_612150 *unk_8600F8::sub_6124A0(const char *a) {
+    if (this->entries == NULL) {
+        return nullptr;
+    }
+
+    int index = get_entry_index(a);
+    
+    if (this->entries[index] == nullptr) {
+        return nullptr;
+    }
+
+    hash_map_entry *entry = this->entries[index];
+    
+    while (entry != nullptr) {
+        char *entryName = entry->name;
+
+        const char* a_p = a;
+        while (*a_p == *entryName) {
+
+            if (*entryName == '\0') {
+                return entry->value;
+            }
+            entryName++;
+            a_p++;
+        }
```

Comment:
Isn't this whole while loop basically `if (std::strcmp(a, entryName) == 0) return entry->value;`?